adr/0001-stac-admin-plugin-system.md

```diff
@@ -0,0 +1,37 @@
+# STAC admin plugin system
+
+    Status: IN REVIEW
+    Deciders: @danielfdsilva @emmanuelmathot @j08lue @oliverroick
+    Date: 2024-08-26
+
+
+## Context and Problem Statement
+
+The form to edit the STAC metadata may differ from instance to instance and from collection to collection. Ideally the form should be easily customizable and extensible to suit the different needs.
+
+## Decision Drivers
+
+The chosen option should allow:
+- Each instance to define what fields the form editor should have;
+- Within an instance, each collection/item to define what fields the form editor should have based on some sort of, possibly dynamic, criteria;
```

Comment:
I think, there should be some dependency on the extensions specified for the catalog/collection/item. You would want to support data editing for the fields specified in the extensions, within reason, as discussed, we want to restrict some fields. But as a general rule, I would aim to support as many fields as we can and the aim is to support the extensions of the catalog/collection/item.

Reply:
I agree. I only meant that, ultimately, it is the instance to define which fields are editable without being forced to anything. The plugins we'd provide support the majority of fields, but if the instances choose to use custom ones, they're free to do so. Maybe this should be written differently to make it clearer? What do you think?

Reply:
Maybe something like this:

> - Each instance controls what fields are available in the editor.
> - We will provide and maintain plugins to that support editing meta data specified by a variety of commonly used STAC extensions. 
> - Through instance-specific plugins, instances can define additional custom fields if required.